x-pack/plugin/sql/src/test/resources/org/elasticsearch/xpack/sql/planner/querytranslator_tests.txt

```diff
@@ -0,0 +1,701 @@
+"params":{"v0":"date","v1":"Z","v2":"YEAR","v3":"LONG"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+OrderByWithCastWithMissingRefs
+SELECT keyword FROM test ORDER BY date::TIME, int LIMIT 5;
+"sort":[{"_script":{"script":{"source":"InternalQlScriptUtils.nullSafeSortString(InternalSqlScriptUtils.cast(InternalQlScriptUtils.docValue(doc,params.v0),params.v1))
+"params":{"v0":"date","v1":"TIME"}
+"type":"string",
+"order":"asc"}},{"int":{"order":"asc","missing":"_last","unmapped_type":"integer"}}]}
+;
+
+
+// CONVERT
+//////////
+GroupByConvertScalar1
+SELECT CONVERT(ABS(EXTRACT(YEAR FROM date)), SQL_BIGINT)
+FROM test
+GROUP BY CONVERT(ABS(EXTRACT(YEAR FROM date)), SQL_BIGINT)
+ORDER BY CONVERT(ABS(EXTRACT(YEAR FROM date)), SQL_BIGINT) NULLS FIRST;
+InternalSqlScriptUtils.cast(InternalSqlScriptUtils.abs(InternalSqlScriptUtils.dateTimeExtract(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)),params.v3)
+"params":{"v0":"date","v1":"Z","v2":"YEAR","v3":"LONG"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+GroupByConvertScalar2
+SELECT EXTRACT(HOUR FROM CONVERT(date, SQL_TIMESTAMP))
+FROM test
+GROUP BY EXTRACT(HOUR FROM CONVERT(date, SQL_TIMESTAMP));
+InternalSqlScriptUtils.dateTimeExtract(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)
+"params":{"v0":"date","v1":"Z","v2":"HOUR_OF_DAY"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+GroupByConvertScalarWithAlias1
+SELECT CONVERT(ABS(EXTRACT(YEAR FROM date)), SQL_BIGINT) as "convert"
+FROM test
+GROUP BY "convert"
+ORDER BY "convert" NULLS FIRST;
+InternalSqlScriptUtils.cast(InternalSqlScriptUtils.abs(InternalSqlScriptUtils.dateTimeExtract(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)),params.v3)
+"params":{"v0":"date","v1":"Z","v2":"YEAR","v3":"LONG"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+GroupByConvertScalarWithAlias2
+SELECT EXTRACT(MINUTE FROM CONVERT(date, SQL_TIMESTAMP)) x
+FROM test
+GROUP BY x;
+InternalSqlScriptUtils.dateTimeExtract(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)
+"params":{"v0":"date","v1":"Z","v2":"MINUTE_OF_HOUR"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+GroupByConvertScalarWithNumericRef
+SELECT CONVERT(ABS(EXTRACT(YEAR FROM date)), SQL_BIGINT)
+FROM test
+GROUP BY 1
+ORDER BY 1 NULLS FIRST;
+InternalSqlScriptUtils.cast(InternalSqlScriptUtils.abs(InternalSqlScriptUtils.dateTimeExtract(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)),params.v3)
+"params":{"v0":"date","v1":"Z","v2":"YEAR","v3":"LONG"}
+"missing_bucket":true
+"value_type":"long","order":"asc"
+;
+
+
+// IN
+/////
+InExpressionWhereClause1
+SELECT * FROM test WHERE keyword IN ('foo', 'bar', 'lala', 'foo', concat('la', 'la'));
+{"terms":{"keyword":["foo","bar","lala"],"boost":1.0}}
+;
+
+InExpressionWhereClause21
```

Comment:
```suggestion
InExpressionWhereClause2
```

Reply:
No worries, I'll fix it. Nice catch, thx!